### ISSUE TITLE
br close silently skips (mislabels 'already closed or not found') an open issue that has an open blocking dependency

### ISSUE THREAD

**Issue by code-of-kai** [Dicklesworthstone/beads_rust#380]
### Summary
`br close <id>` silently refuses to close an **open** issue when that issue has an **open blocking dependency**, and reports the misleading error `Nothing to do: all N issue(s) skipped` / `Hint: All specified issues were already closed or not found` — even though the issue is open (confirmed via `br show` and a direct sqlite read: `status=open, closed_at=NULL`). `--bypass-policy --bypass-reason` does not help. This cost real debugging time because the message claims the issue is closed/missing when it is neither.

### Version
`br 0.2.16`

### Minimal reproduction
```sh
A=$(br create "blocker-A" -t task --ephemeral --silent)
B=$(br create "blocked-B" -t task --ephemeral --silent)
br dep add "$B" "$A" --type blocks        # B is blocked by (still-open) A

br close "$B" -r "done"
# -> Error: Nothing to do: all 1 issue(s) skipped
#    Hint: All specified issues were already closed or not found.

br show "$B"                               # ... but B is OPEN (status: open, closed_at: null)
```
`br close` on an **unblocked** open issue works fine; only the presence of an open `blocks` dependency triggers the skip. Closing A first (removing the open blocker) then lets `br close "$B"` succeed.

### Why this is two bugs
1. **Wrong/misleading diagnostic.** The issue is open and resolvable — `br update <id> --status closed` finds the same issue and refuses with an accurate message (`terminal-state transitions must go through br close ...`, ref #301). But `br close` mis-reports it as "already closed or not found." The skip reason and the message disagree.
2. **Arguable behavior.** Refusing to close a *blocked* issue is questionable: work is often completed out of dependency order, or the blocker is stale. If the refusal is intentional it should be explicit (e.g. `cannot close <id>: blocked by open <blocker-id>; close the blocker or pass --force`) rather than silently dropped as "already closed or not found."

### Suggested fix
In `br close`'s pre-filter, separate "already closed / not found" from "has an open blocking dependency," and either (a) allow the close (a blocked issue can legitimately be complete) or (b) emit an explicit blocked-by error naming the open blocker(s), ideally with a `--force`/`--allow-blocked` opt-out.

**Comment by Dicklesworthstone** [comment]
Thanks for the excellent, precise report — reproduced exactly as described, and you were right on both counts. Fixed in 480a412d.

What was happening: the per-issue skip reason was computed correctly internally ("blocked by: <id>"), but when *every* requested issue was skipped, the terminal error collapsed to a generic `Nothing to do: all N issue(s) skipped`, and the NOTHING_TO_DO hint unconditionally claimed "All specified issues were already closed or not found" — flatly wrong for a dependency block. That is the message you saw, and it is exactly what sent you into sqlite forensics.

What changed (in 480a412d, on `main`):

1. The blocked-skip reason now names the open blocker(s) **and** the way out:
   `Skipped bx-2: blocked by: bx-1 — close the open blocker(s) first, or use --force to close anyway`
2. The terminal error now embeds the real per-issue reasons (up to five, then `+N more`):
   `Error: Nothing to do: all 1 issue(s) skipped — bx-2: blocked by: bx-1 — close the open blocker(s) first, or use --force to close anyway`
3. The hint is now derived from what actually happened. Dependency blocks get:
   `Hint: Skipped issue(s) have open blocking dependencies. Close the blockers first, or re-run with --force to close anyway.`
   The old "already closed or not found" wording only appears when that is genuinely the case.

On your point 2 (arguable behavior): refusing to close a blocked issue is intentional close-ordering policy, but the opt-out you asked for already exists — `br close <id> --force` ("Close even if blocked by open dependencies") closes a blocked issue directly. The bug was purely that nothing in the failure path ever told you so; now both the skip warning and the hint point at it.

Also added an e2e test that drives the human-mode failure path and asserts the error names the blocker, points at `--force`, and never claims the issue is closed/missing. The fix will ship in the next release (current published release 0.2.16/0.2.17 predates it).
